src/rt/ds/stackarray.h

```diff
@@ -0,0 +1,63 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+
+/**
+ * @brief A stack allocated array.
+ *
+ * Due to portability allocates a largeish array on the stack, if this array is
+ * not big enough then dynamically allocates something of the correct size. This
+ * is done to avoid the need for a dynamic allocation in the common case.
+ */
+template<typename T>
+class StackArray
+{
+  // Size of the stack allocated array.
+  static constexpr size_t Size = 128;
+
+  // Stack allocated array.
+  T main[Size];
```

Comment:
I am not sure about the exact details, but I feel like the contents of this could end up as being uninitialized. Looking at an example on CompilerExplorer, the LLVM IR shows "undef" for reading it, but gets a codegen of 0 so it would not show up during execution.

Reply:
I have switched to untyped memory and added code to perform initialization.